File: general programming/java/programs/src/hello/prime_number.java
package hello;

import java.util.Scanner ;

public class prime_number {
    public static void main(String []args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter a number: ");
        int noToBeChecked = sc.nextInt();
        //int m = noToBeChecked/2; //or sqrt(noToBeChecked)
        int m = (int)Math.sqrt(noToBeChecked);
        Boolean flag=false;
        if(noToBeChecked==0 || noToBeChecked==1) {
            System.out.println(noToBeChecked + " is not a prime number");
        }
        for(int i=2; i<m; i++) {
            if(noToBeChecked%i == 0) {
                flag = true;
                break;
            }
        }
        if(flag.equals(true)) {
            System.out.println(noToBeChecked + " is not a prime number");
        } else {
            System.out.println(noToBeChecked + " is prime number");
        }
    }
}

File: general programming/c++/prime-number.cpp
#include<bits/stdc++.h>

using namespace std;

int main() {
    int i, n, sqrtOfn, flag=0;

    printf("Enter a number: ");
    scanf("%d", &n);
    sqrtOfn = sqrt(n);
    if(n==0 || n==1) {
        flag = 0;
    }else {
        for (i=2; i<=sqrtOfn; i++) {
            if (n%i==0) {
                flag =1;
                break;
            }
        }
    }
    if (flag==1) {
        printf("%d is not prime number ", n);
    }else {
        printf("%d is prime number ", n);
    }

    return 0;
}